api/v1alpha1/perconaservermysql_types.go

```diff
@@ -330,18 +333,23 @@ func (cr *PerconaServerMySQL) CheckNSetDefaults(serverVersion *platform.ServerVe
 	return nil
 }
 
-func reconcileVol(v *VolumeSpec) *VolumeSpec {
-	if v == nil {
-		v = &VolumeSpec{}
-	}
-
-	if v.EmptyDir == nil && v.HostPath == nil && v.PersistentVolumeClaim == nil {
-		v.PersistentVolumeClaim = &corev1.PersistentVolumeClaimSpec{}
+func reconcileVol(v *VolumeSpec) (*VolumeSpec, error) {
+	if v == nil || v.EmptyDir == nil && v.HostPath == nil && v.PersistentVolumeClaim == nil {
+		return nil, errors.New("volumeSpec and it's internals should be specified")
+	}
+	if v.PersistentVolumeClaim == nil {
+		return nil, errors.New("pvc should be specified")
+	} else {
+		_, limits := v.PersistentVolumeClaim.Resources.Limits[corev1.ResourceStorage]
+		_, requests := v.PersistentVolumeClaim.Resources.Requests[corev1.ResourceStorage]
+		if !(limits || requests) {
+			return nil, errors.New("pvc's resources.limits[storage] or resources.requests[storage] should be specified")
+		}
```

Comment:
I don't think we need this`else` here, the function will return if `v.PersistentVolumeClaim == nil`

Reply:
fixed https://github.com/percona/percona-server-mysql-operator/pull/76/commits/47a57c3a3f6f9ec616ae1f4bc15f95d0952304ec